tools/update.php

```diff
@@ -0,0 +1,705 @@
+<?php
+
+/**
+ * After running this script, a user should be able to go from an outdated LORIS
+ * instance to the newest install without issue.
+ *
+ * This includes having the latest minor release files, third-party packages
+ * (apt, npm, composer), PHP Version, etc.
+ *
+ * ## Details
+ *
+ * Specifically:
+ *
+ *- [x] **Creates a back up of `$LORIS_root`**
+ *
+ *- [x] (If not on a development instance) **Downloads the release source code
+ *      and overwrites `$LORIS_root`**
+ *
+ *- [x] **Ensures that apt dependencies are installed and up-to-date**
+ *
+ *- [x] **Runs 3rd-party package managers** (npm and composer)
+ *
+ *- [x] **Outputs a list of SQL patches** for the user to apply manually
+ *      (including development patches if on a LORIS dev instance)
+ *
+ *- [x] **Provides a lot of output** to assist the user in what's going on.
+ *
+ * Assumptions:
+ *
+ * As of now the script always runs in "Report-Only" mode, meaning
+ * SQL patches are not applied by the script. However, if desired,
+ * this functionality can be added easily by creating a command-line arg and
+ * calling the apply function with `$report_only_mode = false`. I chose not to
+ * include it as this is a potentially risky process.
+ *
+ * As of now there is no "rollback" feature. `rsync` is used to overwrite the
+ * LORIS-root source code from the unzipped release folder. A backup is created,
+ * but should something go wrong with `rsync`, the user will need to restore the
+ * source code files manually. In this case, an error message is displayed and
+ * the script will exit.
+ *
+ * The function to retrieve patches is heavily dependent on the way we organize
+ * and name SQL files. If the directory structure changes, this functionality
+ * is likely to break.
+ *
+ * @category Update
+ * @package  Tools
+ * @author   John Saigle <john.saigle@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+require_once __DIR__ . "/../vendor/autoload.php";
+require_once "../php/libraries/Database.class.inc";
+require_once "../php/libraries/NDB_Config.class.inc";
+
+error_reporting(E_ALL);
+$minimum_php = 7; //TODO: Update this value as time passes
+if (PHP_MAJOR_VERSION < $minimum_php) {
+    die("ERROR: {$argv[0]} and LORIS require PHP $minimum_php or higher.");
+}
+echo 'This script will prompt for superuser privileges,'
+    . ' as they are needed to e.g. update apt packages.' . PHP_EOL;
+echo 'You may wish to review code changes tagged with "Caveat For '
+    . 'Existing Projects" as they may include changes to your workflow. '
+    . PHP_EOL
+    // @codingStandardsIgnoreStart
+    . "\tSee: https://github.com/aces/Loris/pulls?q=is%3Apr+label%3A%22Caveat+for+Existing+Projects%22+is%3Amerged"
+    // @codingStandardsIgnoreEnd
+    . PHP_EOL . PHP_EOL;
+sleep(3);
+main();
```

Comment:
I think it would be best if the script required a `-confirm` to actually update, and without it would just spew out some help text (and the newest release version # so users can be sure they want to update), just so this script is never accidentally started

Reply:
There is now a confirm flag present.